### PR TITLE
arelle integration pull request for Edgar 20.3

### PR DESCRIPTION
**Pull request by hefischer**
changes to integrate with EdgarRenderer, POST uploads and zip downloads via REST API; tolerate nonfatal instance errors in partially-constructed instances

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: iXBRLViewerPlugin/__init__.py [hunk 1]
```diff
@@ -29,37 +54,50 @@ def iXBRLViewerCommandLineOptionExtender(parser, *args, **kwargs):
...
+        out = getattr(options, 'saveViewerFile') or kwargs.get("responseZipStream")
     if out:
-        viewerBuilder = IXBRLViewerBuilder(cntlr.modelManager.modelXbrl)
+        viewerBuilder = IXBRLViewerBuilder(modelXbrl)
         iv = viewerBuilder.createViewer(scriptUrl=options.viewerURL)
-        iv.save(out)
+        iv.save(out, outBasenameSuffix=VIEWER_BASENAME_SUFFIX, outzipFilePrefix=VIEWER_BASENAME_SUFFIX)
+
+
+except Exception as ex:
```
Comment (brettkail-wk) [inline iXBRLViewerPlugin/__init__.py:72]: This indentation looks off.

File: iXBRLViewerPlugin/__init__.py [hunk 2]
```diff
@@ -70,15 +108,33 @@ def commandLineRun(*args, **kwargs):
...
     'description': "iXBRL Viewer creator",
     'license': 'License :: OSI Approved :: Apache License, Version 2.0 (Apache-2.0)',
     'author': 'Paul Warren',
     'copyright': 'Copyright :: Workiva Inc. :: 2019',
-    'imports': ["./iXBRLViewer.py"],
     'CntlrCmdLine.Options': commandLineOptionExtender,
     'CntlrCmdLine.Xbrl.Run': commandLineRun,
-    'CntlrWinMain.Menu.Tools': menuExtender,
+    'CntlrWinMain.Menu.Tools': toolsMenuExtender,
+'CntlrWinMain.Menu.View': viewMenuExtender,
```
Comment (brettkail-wk) [inline iXBRLViewerPlugin/__init__.py:138]: This indentation also looks off.

**Comment by hefischer** [comment]
Git won’t authorize me to use git to push from local to git hub, and resulting manual GUI merge is causing strange merge problems, sorry.  Something wrong with my 2 factor authentication setup.

   Herm

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by sagesmith-wf — DISMISSED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by sagesmith-wf — APPROVED** [review]

**Review by austinmatherne-wk — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Unit tests pass successfully. Built locally and verified no regressions.

@Workiva/release-management-p

**Comment by dancleereman-wf** [comment]
RM +1

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**